### PR TITLE
Shared keys

### PR DESCRIPTION
**Pull request by doomedraven**
define all keys under yeti, but don't expose apikeys/creds to the rest of users, aka non admin users :), but users can overwrite settings with their options

### CODE REVIEW

File: core/web/api/analytics.py [hunk 1]
```diff
@@ -1,17 +1,24 @@
...
 from core import analytics
+from core.config.config import yeti_config
 from core.analytics_tasks import schedule
 from core.web.api.api import render
 from core.web.helpers import get_object_or_404
 from core.web.helpers import requires_permissions
 
+shared_keys = False
+if hasattr(yeti_config, "shared_keys"):
+    shared_keys = yeti_config.shared_keys.enabled
```
Comment (tomchop) [inline core/web/api/analytics.py]: I think you can do `shared_keys = yeti_config.get("shared_keys", "enabled", False)`

File: core/web/api/analytics.py [hunk 2]
```diff
@@ -78,13 +85,19 @@ class OneShotAnalytics(CrudApi):
...
+                logging.info(e)
+
         for obj in self.objectmanager.objects.all():
             info = obj.info()
 
             info['available'] = True
-            if hasattr(
-                    obj,
-                    'settings') and not current_user.has_settings(obj.settings):
+            if hasattr(obj, 'settings') and (not current_user.has_settings(obj.settings) and \
```
Comment (tomchop) [inline core/web/api/analytics.py]: please split into boolean variables before evaluation, this is very hard to read

**Comment by doomedraven** [comment]
done

File: core/web/api/analytics.py (hunk 1 above)
Comment (doomedraven) [reply core/web/api/analytics.py]: done

**Pull request closed without merging**